bridges/VkBridge.php

```diff
@@ -49,7 +52,372 @@ public function getName()
+                if ($attachment['type'] == 'photo') {
+                    // JPEG, PNG photos
+                    // GIF in vk is a document, so, not handled as photo
+                    $photo = e($this->getImageURLWithLargestWidth($attachment['photo']['sizes']));
+                    $text = e($attachment['photo']['text']);
+                    $ret .= "<p><img src='{$photo}' alt='{$text}'></p>";
+                } elseif ($attachment['type'] == 'doc' and $attachment['doc']['ext'] == 'gif') {
+                    // GIF docs
+                    $url = e($attachment['doc']['url']);
+                    $ret .= "<p><img src='{$url}'></p>";
+                } elseif ($attachment['type'] == 'link') {
+                    // links
+                    $url = e($attachment['link']['url']);
+                    $url = str_replace('https://m.vk.com', 'https://vk.com', $url);
+                    $title = e($attachment['link']['title']);
+                    if (isset($attachment['link']['photo'])) {
+                        $photo = $this->getImageURLWithLargestWidth($attachment['link']['photo']['sizes']);
+                        $ret .= "<p><a href='{$url}'><img src='{$photo}' alt='{$title}'><br>{$title}</a></p>";
+                    } else {
+                        $ret .= "<p><a href='{$url}'>{$title}</a></p>";
+                    }
+                } elseif ($attachment['type'] == 'note') {
+                    // notes
+                    $title = e($attachment['note']['title']);
+                    $url = e($attachment['note']['view_url']);
+                    $ret .= "<p><a href='{$url}'>{$title}</a></p>";
+                } elseif ($attachment['type'] == 'poll') {
+                    // polls
+                    $question = e($attachment['poll']['question']);
+                    $vote_count = $attachment['poll']['votes'];
+                    $answers = $attachment['poll']['answers'];
+                    $ret .= "<p>Poll: {$question} ({$vote_count} votes)<br />";
+                    foreach ($answers as $answer) {
+                        $text = e($answer['text']);
+                        $votes = $answer['votes'];
+                        $rate = $answer['rate'];
+                        $ret .= "* {$text}: {$votes} ({$rate}%)<br />";
+                    }
+                    $ret .= '</p>';
+                } elseif (!in_array($attachment['type'], ['video', 'audio', 'doc'])) {
+                    $ret .= "<p>Unknown attachment type: {$attachment['type']}</p>";
+                }
+            }
+        }
+
+        return $ret;
+    }
+
+    protected function getImageURLWithLargestWidth($items) {
+        usort($items, function($a, $b) {
+            return $b['width'] - $a['width'];
+        });
+        return $items[0]['url'];
+    }
+
     public function collectData()
+    {
+        if ($this->getOption('access_token')) {
+            $r = $this->api('wall.get', [
+                'domain' => $this->getInput('u'),
```

Comment:
seems that `domain` arg can be used only if group has short link. Maybe it's better to use `owner_id`